aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -217,12 +238,20 @@ void TensorIterator::compute_types() {
           op.dtype = op.tensor.scalar_type();
         } else {
           op.device = common_device;
-          op.dtype = common_dtype;
+          if (compute_common_dtype_only_for_inputs && op.is_output) {
+            op.dtype = op.tensor.scalar_type();
+          } else {
+            op.dtype = common_dtype;
+          }
         }
       }
 
-      validate_dtype(op, common_dtype, ninputs());
-      maybe_promote_common_dtype(op, common_dtype);
+      if (compute_common_dtype_strategy_ == CommonDTypeStrategy::COMPUTE_ALL) {
+        validate_dtype(op, common_dtype, ninputs());
```

Comment:
The following lines are no longer hit:
https://github.com/pytorch/pytorch/blob/a14e88454690645d57e056f45829331df7eef611/aten/src/ATen/native/TensorIterator.cpp#L160-L162 . Is this important? I don't have enough context to understand what `validate_types` does.

Reply:
I changed this to: !compute_common_dtype_only_for_inputs